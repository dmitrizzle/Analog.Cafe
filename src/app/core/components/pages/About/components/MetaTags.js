import { Helmet } from "react-helmet"
import React from "react"

import { makeFroth } from "../../../../../utils"

export default props => {
  return (
    <Helmet>
      <title>{props.metaTitle}</title>
      <meta name="description" content={props.metaDescription} />
      <meta
        property="og:image"
        content={
          makeFroth({
            src: "image-froth_669120_c34babc2fb974c8d9f03249bea647401",
            size: "m"
          }).src
        }
      />
      <meta property="og:title" content={props.metaTitle} />
      <meta property="og:description" content={props.metaDescription} />
    </Helmet>
  )
}
