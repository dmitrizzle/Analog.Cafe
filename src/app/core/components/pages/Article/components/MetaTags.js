import { Helmet } from "react-helmet"
import React from "react"

import { makeFroth } from "../../../../../utils"

export default props => {
  return (
    <Helmet>
      <title>{props.article.title}</title>
      <meta name="description" content={props.article.summary} />
      <meta property="og:title" content={props.article.title} />
      <meta property="og:description" content={props.article.summary} />
      {props.article.poster && (
        <meta
          property="og:image"
          content={makeFroth({ src: props.article.poster, size: "m" }).src}
        />
      )}
    </Helmet>
  )
}
