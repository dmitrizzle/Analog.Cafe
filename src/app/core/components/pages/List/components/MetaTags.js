import { Helmet } from "react-helmet"
import React from "react"

export default props => {
  return (
    <Helmet>
      <title>{props.renderedListTitle}</title>
      <meta name="description" content={props.renderedListMeta.description} />
      <meta property="og:title" content={props.renderedListTitle} />
      <meta
        property="og:description"
        content={props.renderedListMeta.description}
      />
    </Helmet>
  )
}
