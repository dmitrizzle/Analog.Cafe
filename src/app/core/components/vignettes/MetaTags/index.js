import { Helmet } from "react-helmet"
import React from "react"

import { HOST_PROD, HOST_PROTOCOL } from "../../../../constants"
import { ROUTE_URL_ARTICLES } from "../../../constants/routes-article"
import { getISODatestamp } from "../../../utils/messages-"
import { getLeadAuthorObject } from "../../../utils/messages-author"
import { makeFroth } from "../../../../utils"

export default props => {
  const image = makeFroth({ src: props.metaImage, size: "m" }).src
  const schema = {
    type: "application/ld+json",
    innerHTML: JSON.stringify({
      "@context": "http://schema.org",
      "@type": "Article",
      mainEntityOfPage: props.metaSlug
        ? {
            "@type": "WebPage",
            "@id": `${ROUTE_URL_ARTICLES}/${props.metaSlug}`
          }
        : undefined,
      image: image
        ? {
            "@type": "ImageObject",
            url: image
          }
        : undefined,
      headline: props.metaTitle ? props.metaTitle : undefined,
      publisher: {
        "@type": "Organization",
        name: "Analog.Cafe",
        logo: {
          "@type": "ImageObject",
          url: `${HOST_PROTOCOL + HOST_PROD}/icon-512x512.png`,
          width: "512",
          height: "512"
        }
      },
      author:
        props.metaAuthors && props.metaAuthors[0].name
          ? {
              "@type": "Person",
              name: getLeadAuthorObject(props.metaAuthors).name
            }
          : undefined,
      datePublished: props.metaPostDate
        ? getISODatestamp(props.metaPostDate)
        : undefined,
      dateModified: props.metaEditDate
        ? getISODatestamp(props.metaEditDate)
        : props.metaPostDate
          ? props.metaPostDate
          : undefined
    })
  }

  return (
    <Helmet {...(props.metaArticleSchema ? { script: [schema] } : {})}>
      {props.metaTitle && <title>{props.metaTitle}</title>}
      {props.metaTitle && (
        <meta property="og:title" content={props.metaTitle} />
      )}
      {props.metaDescription && (
        <meta name="description" content={props.metaDescription} />
      )}
      {props.metaDescription && (
        <meta property="og:description" content={props.metaDescription} />
      )}
      {props.metaImage && (
        <meta
          property="og:image"
          content={makeFroth({ src: props.metaImage, size: "m" }).src}
        />
      )}
    </Helmet>
  )
}
