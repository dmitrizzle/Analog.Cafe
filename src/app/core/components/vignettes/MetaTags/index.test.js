import "jest-styled-components"

import { shallow } from "enzyme"
import React from "react"

import MetaTags from "./"

it("Render MetaTags, matches snapshot", () => {
  const element = shallow(<MetaTags />)
  expect(element).toMatchSnapshot()
})
