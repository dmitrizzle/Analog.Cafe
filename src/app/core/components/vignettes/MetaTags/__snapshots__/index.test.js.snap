// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MetaTags, matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      "defer": true,
      "encodeSpecialCharacters": true,
    },
    "ref": null,
    "rendered": Array [
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ],
        "defer": true,
        "encodeSpecialCharacters": true,
      },
      "ref": null,
      "rendered": Array [
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
